src/Lime.Protocol/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Lime.Protocol
+{
+    /// <summary>
+    /// Utilities extensions for the <c>IEnumerable</c> interface
+    /// </summary>
+    public static class IEnumerableExtensions
+    {
+        /// <summary>
+        /// Creates a new <c>DocumentCollection</c> object using a given <paramref name="documentEnumerable"/>
+        /// </summary>
+        /// <returns>The document collection.</returns>
+        /// <param name="documentEnumerable"><c>IEnumerable</c> of <c>Document</c>.</param>
+        /// <param name="size">The number of <c>Documents</c> in the collection</param>
+        /// <typeparam name="T">Subtype of the <c>Document</c>'s contained in the <c>IEnumerable</c></typeparam>
+        public static DocumentCollection ToDocumentCollection<T>(this IEnumerable<T> documentEnumerable, int size = 0) where T : Document
+        {
+            return new DocumentCollection
+            {
+                Items = documentEnumerable.ToArray(),
+                ItemType = documentEnumerable.FirstOrDefault().GetMediaType(),
```

Comment:
`FirstOrDefault` will return null if documentEnumerable has no elements, causing an NullReferenceException when trying to call GetMediaType.
If this is actually the expected scenario, I recommend checking earlier if documentEnumerable is null or its Count method returns 0 and throw an ArgumentException.